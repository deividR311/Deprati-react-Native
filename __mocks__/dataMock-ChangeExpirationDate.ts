export const DataMock_ChangeExpirationDate = {
  TEXT_NEWDATE: 'Selecciona el día',
  infoCreditBalance: {
    clientName: 'PATRICIA L. DE UGALDE',
    cardNumber: '1xxxxx6',
    affiliateDate: '2000',
    totalAmount: '5.500,00',
    amountSpent: '724,35',
    aditional: '00',
    availableAmount: '4.775,65',
    pastDueDays: 0,
    nextPayDay: 'Abril 14 de 2023',
    nextAmountToPay: '105,08',
    numeroTarjetaAdicionalDisplay: '14XXX86 00',
    clientFirstName: 'Prueba'
  },
  dataCurrentDate: {
    cuttingDay: 14,
    dueDate: '20230514',
    day: '14'
  },
  dataChangeDate: null,
  selectedDay: '2',
  isAcceptTerms: false,
  showWarning: false,
  isLoadingChangeDate: false,
  isLoadingCurrentDate: false,
  showSucces: false,
  showErrorSucces: false,
  expirationDateContent: {
    legend:
      'No puedes cambiar tu fecha máxima de pago si tienes una deuda pendiente y <b>este cambio se hará efectivo a partir del próximo mes.</b>',
    popup_title:
      '¿Estás seguro de cambiar la fecha máxima de pago de tu crédito De Prati?',
    popup_paragraphs: [
      'Ten en cuenta:',
      '1. Esta acción no se puede deshacer y el cambio se hará efectivo a partir de tu próximo mes.',
      '2. Si haces el cambio de fecha <b>deberás pagar la cuota del mes en curso</b> y este cambio se hará efectivo a partir del próximo mes.'
    ],
    popup_accept_action: 'SI, ESTOY DE ACUERDO',
    popup_cancel_action: 'NO, CANCELAR'
  },
  selectRef: {
    current: {
      reset: jest.fn(),
      openDropdown: jest.fn(),
      closeDropdown: jest.fn()
    }
  },
  setSelectedDay: jest.fn(),
  setIsAcceptTerms: jest.fn(),
  handleChangeDate: jest.fn(),
  setShowWarning: jest.fn(),
  handleCloseSuccess: jest.fn(),
  handleCloseError: jest.fn(),
  handleCloseWarning: jest.fn()
};
