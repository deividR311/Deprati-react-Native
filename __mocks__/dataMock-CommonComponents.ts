export const DataMock_CustomProps = {
  url: 'url',
  style: {},
  content: 'qwerty',
  uid: 'uid',
  typeCode: 'qwerty',
  modifiedTime: 'qwerty',
  name: 'qwerty',
  container: true,
  rotatingImagesComponentsList: 'DepratiBrandsRotatingImages',
  childrenComponentsList: ['DepratiBrandsRotatingImages'],
  childrenComponents: {
    DepratiBrandsRotatingImages: {
      uid: 'DepratiBrandsRotatingImages',
      typeCode: 'RotatingImagesComponent',
      modifiedTime: '2023-04-28T12:59:58+0000',
      name: 'DePrati Dior Mobile Carousel',
      container: 'false',
      resolutionEnum: 'MOBILE',
      effect: 'FADE',
      banners: 'DiorBannerComponent2 BannerCarouselDiorVideo_1',
      timeout: '6',
      childrenComponentsList: [
        'DiorBannerComponent2',
        'BannerCarouselDiorVideo_1'
      ]
    }
  }
};

export const DataMock_BillingAddressComponent = {
  style: {},
  onSelected: () => {},
  emptyStyle: {},
  showTitle: true,
  enableHandleEnableButton: true,
  onIsEnable: () => {},
  paymentAddress: {
    defaultAddress: false,
    email: 'string',
    firstName: 'string',
    formattedAddress: 'string',
    id: 'string',
    line1: 'string',
    line2: 'string',
    phone: 'string',
    phonePreffix: 'string',
    shippingAddress: false,
    visibleInAddressBook: false
  },
  paymentCredit: {
    showAlerAddressPayment: false,
    setShowAlerAddressPayment: () => {}
  },
  styleSecundaryButton: false
};

export const DataMock_useBillingAddress = {
  addressSelected: {
    defaultAddress: false,
    firstName: 'firstName',
    id: 'qwertystring',
    line1: 'qwerty',
    line2: 'qwerty',
    phone: 'qwerty',
    phonePreffix: 'qwerty'
  },
  isLoadingByDeleting: false,
  addressPaymentList: {
    default: null,
    others: [],
    existsAddress: false,
    loading: false
  },
  isLoading: false,
  buttonChangeData: false,
  handleChangeData: jest.fn(),
  changeAddress: {
    handleSelected: jest.fn(),
    addressSelected: undefined
  },
  handleDelete: jest.fn(),
  handleAddEdit: jest.fn(),
  handleBillingAddressList: () => {
    return [
      {
        defaultAddress: false,
        email: 'string',
        firstName: 'string',
        formattedAddress: 'string',
        id: 'string',
        line1: 'string',
        line2: 'string',
        phone: 'string',
        phonePreffix: 'string',
        shippingAddress: false,
        visibleInAddressBook: false
      }
    ];
  }
};

export const DataMock_IncreaseCreditLimit = {
  rectangularSize: 232,
  value: 1000,
  maxValue: 2000,
  onChangeCreditLimit: jest.fn()
};

export const DataMock_Charts = {
  constentChart: {
    availableAmount: 100,
    amountSpent: 20,
    totalAmount: 200
  },
  ownerFullname: 'Custom customer customization',
  invertValues: true,
  showDisplayCard: true
};
