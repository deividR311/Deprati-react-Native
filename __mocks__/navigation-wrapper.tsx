import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
