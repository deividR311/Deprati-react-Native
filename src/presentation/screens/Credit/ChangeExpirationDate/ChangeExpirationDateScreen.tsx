//libs
import React, { FC, useMemo } from 'react';
import { ScrollView, Text, View } from 'react-native';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { COLORS, FONTS_SIZES, NAV } from '../../../../application/common';
//components
import TemplatePage from '../../../common-components/template-page';
import { CreditNavigationProps } from '../../../navigation/credit';
import { ClientInfoCard } from '../credit-limit-increase/components/client-info-card';
import CheckboxComp from '../../../common-components/checkboxs';
import TextAndLink from '../../SignUp/components/TextAndLink';

import { Popup } from '../../../common-components/poppup';
import { MainButton } from '../../../common-components/buttons/Button';

import TextWarningComponent from './components/TextWarningComponent';
import TextSuccessComponent from './components/TextSuccessComponent';
import { SkeletonMonthInfo } from './components/SkeletonChangeExpirationDate';
import SelectInput from '../../../common-components/inputs/SelectInput';
import { AlertCard } from '../../../common-components/alert-card/AlertCard';
import { CircularWarningIcon } from '../../../../../assets/icons/CircularWarningIcon';
//hooks
import useChangeExpirationDate from './hook/useChangeExpirationDate.hook';
import useOpenLinkInApp from '../../../../application/common/hooksCommons/useOpenLinkInApp';

//utils
import { styles, stylesPopup } from './styleChangeExpirationDate';
import formatTextByTags from '../../../../application/utils/formatTextByTags';
import { getUrlCreditTerms } from '../../../../application/utils/urls';
import { PopupWhatsapp } from '../../../common-components/poppup/poppupWhatsapp';
import { useTranslation } from 'react-i18next';

const TEXT_DESCRIPTION =
  'No puedes cambiar tu fecha m√°xima de pago si tienes una deuda pendiente y este <b>cambio se har√° efectivo a partir del pr√≥ximo mes.</b>';

const TITLE_ALERT =
  '¬øEst√°s seguro de cambiar la fecha m√°xima de pago de tu cr√©dito De Prati?';

const TITLE_SUCCESS =
  '¬°La solicitud de cambio de fecha ha sido enviada con √©xito! \\n Ten en cuenta que tu nueva fecha de vencimiento es el {0} de cada mes\\ny se har√° efectiva a partir de {1} de {2}';

const TITLE_ERROR =
  '-No se pudo generar la solicitud de cambio de fecha de vencimiento, Volver a Intentar';
export const ChangeExpirationDateScreen: FC<
  ChangeExpirationDateScreenProps
> = props => {
  const { t } = useTranslation();
  const openUrl = useOpenLinkInApp();
  const {
    TEXT_NEWDATE,
    infoCreditBalance,
    dataCurrentDate,
    dataChangeDate,
    selectedDay,
    setSelectedDay,
    isAcceptTerms,
    setIsAcceptTerms,
    handleChangeDate,
    showWarning,
    setShowWarning,
    isLoadingChangeDate,
    isLoadingCurrentDate,
    showSucces,
    handleCloseSuccess,
    showErrorSucces,
    handleCloseError,
    expirationDateContent,
    handleCloseWarning,
    selectRef,
    showModalChangeDate,
    errorChangeDate,
    handleCloseModal
  } = useChangeExpirationDate();

  const calendarDayNumber = useMemo(() => {
    // return getNextMonthDays().map(v => ({
    //   label: v.label,
    //   value: v.value.getDate().toString(),
    // }))

    /** @todo IMPOERTANTE!!! */
    // üëÜ Utiliza esta üëÜ si se debe calcular por el mes
    // O utiliza esta üëá si solo es quemar los 30 dias

    return new Array(30).fill(1).map((v, i) => ({
      label: (v + i).toString(),
      value: (v + i).toString()
    }));
  }, []);

  const successMessage = useMemo(() => {
    const message =
      dataChangeDate?.message?.split('\\n') ?? TITLE_SUCCESS.split('\\n');
    const title = message[0];

    const content = [...message];
    content.shift();

    return {
      title,
      content: content.join(`\n`)
    };
  }, [dataChangeDate?.message]);

  return (
    <TemplatePage loading={false} error={false} skeleton={null}>
      <ScrollView>
        <View style={styles.container}>
          <ClientInfoCard
            clientName={infoCreditBalance?.clientName ?? ''}
            accountNumber={
              infoCreditBalance?.numeroTarjetaAdicionalDisplay ?? ''
            }
            clientSince={infoCreditBalance?.affiliateDate ?? ''}
          />
          <View
            style={{
              backgroundColor: COLORS.GRAYDARK20,
              padding: 12,
              marginVertical: 16
            }}>
            <Text style={[styles.subTitle]}>Fecha de pago actual</Text>
            {isLoadingCurrentDate ? (
              <SkeletonMonthInfo />
            ) : (
              <Text style={[styles.title]}>
                {`${dataCurrentDate?.cuttingDay ?? ''} de cada mes`}
              </Text>
            )}
          </View>

          <>
            <Text style={[styles.subTitle, { marginBottom: 10 }]}>
              Cambia la fecha m√°xima de pago
            </Text>
            <Text style={[styles.title]}>
              El d√≠a que selecciones aplicar√° para todos tus siguientes pagos de
              tu cr√©dito De Prati
            </Text>
          </>

          <View style={styles.viewInput}>
            <SelectInput
              selectRef={selectRef}
              useOnlyPlaceHolder
              autoPlaceholderMessage={false}
              label="Selecciona el d√≠a"
              items={calendarDayNumber}
              onChange={(day: string | number) => setSelectedDay(day)}
              value={selectedDay}
            />
          </View>

          <AlertCard
            icon={<CircularWarningIcon />}
            text={
              <Text style={[styles.description]}>
                {formatTextByTags({
                  text: expirationDateContent?.legend ?? TEXT_DESCRIPTION
                })}
              </Text>
            }
          />
        </View>

        <CheckboxComp
          onPress={() => setIsAcceptTerms(!isAcceptTerms)}
          color={COLORS.BRAND}
          status={isAcceptTerms ? 'checked' : 'unchecked'}
          styleContainer={styles.check}
          textStyle={{
            fontSize: FONTS_SIZES.label,
            color: COLORS.DARK,
            width: '100%'
          }}
          label={
            <TextAndLink
              textLinkStyle={styles.terms}
              text={t('acceptThe')}
              textLink={t('termsAndConditions')}
              onPress={() => {
                openUrl(getUrlCreditTerms());
              }}
            />
          }
        />
        <View style={styles.viewButton}>
          <MainButton
            disabled={
              !isAcceptTerms ||
              selectedDay === TEXT_NEWDATE ||
              isLoadingChangeDate ||
              !dataCurrentDate?.cuttingDay
            }
            showActivityIndicator={isLoadingChangeDate}
            style={styles.button}
            title={'CAMBIAR FECHA'}
            onPress={() => setShowWarning(true)}
          />
        </View>
        <View style={{ height: 80, width: '100%' }} />
      </ScrollView>
      <Popup
        visible={showWarning}
        doubleButton={true}
        title={expirationDateContent?.popup_title ?? TITLE_ALERT}
        textComponent={() => (
          <TextWarningComponent content={expirationDateContent} />
        )}
        buttonText={expirationDateContent?.popup_cancel_action}
        secondButtonText={
          expirationDateContent?.popup_accept_action ?? 'SI, ESTOY DE ACUERDO'
        }
        contentStyle={stylesPopup.contentStyle}
        titleStyle={stylesPopup.titleStyle}
        containerButtonsStyle={stylesPopup.containerButtonsStyle}
        buttonTextStyle={stylesPopup.buttonTextStyle}
        secondButtonTextStyle={stylesPopup.secondButtonTextStyle}
        closeAction={handleCloseWarning}
        buttonAction={handleChangeDate}
      />
      <Popup
        visible={showSucces}
        closeAction={handleCloseSuccess}
        icon={'check-circle'}
        iconColor={COLORS.GREENOK}
        title={successMessage.title}
        textComponent={() => (
          <TextSuccessComponent content={successMessage.content} />
        )}
        showCloseButton={true}
        buttonAction={handleCloseSuccess}
        buttonText="CERRAR"
        buttonType="full"
        contentStyle={stylesPopup.contentSuccessStyle}
      />
      <Popup
        visible={showErrorSucces}
        closeAction={handleCloseError}
        icon={'error'}
        iconColor={COLORS.BRAND}
        title={dataChangeDate?.message?.split('\n')[0] ?? TITLE_ERROR}
        showCloseButton={true}
        hideButton={true}
      />
      <PopupWhatsapp
        visible={showModalChangeDate}
        title={errorChangeDate?.data.message ?? errorChangeDate?.data.Message}
        onClose={handleCloseModal}
      />
    </TemplatePage>
  );
};

export interface ChangeExpirationDateScreenProps
  extends NativeStackScreenProps<
    CreditNavigationProps,
    NAV.CreditChangeExpirationDate
  > {}
