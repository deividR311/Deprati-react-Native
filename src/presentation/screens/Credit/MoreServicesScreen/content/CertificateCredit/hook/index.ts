export { default as useCertificateCredit } from './useCertificateCredit.hook';
