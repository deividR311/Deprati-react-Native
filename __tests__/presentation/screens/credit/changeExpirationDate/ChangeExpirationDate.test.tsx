import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../../__mocks__/navigation-wrapper';
import { ChangeExpirationDateScreen } from '../../../../../src/presentation/screens/Credit/ChangeExpirationDate/ChangeExpirationDateScreen';
import { useChangeExpirationDate } from '../../../../../src/presentation/screens/Credit/ChangeExpirationDate/hook/useChangeExpirationDate.hook';
import { DataMock_ChangeExpirationDate } from '../../../../../__mocks__/dataMock-ChangeExpirationDate';

jest.mock(
  '../../../../../src/presentation/screens/Credit/ChangeExpirationDate/hook/useChangeExpirationDate.hook'
);

jest.mock('react-native-select-dropdown', () => {
  const { View } = require('react-native');
  const MockSelectDropdown = (props: any) => {
    return <View>{props.children}</View>;
  };

  return MockSelectDropdown;
});

describe('ChangeExpirationDate', () => {
  it('renders correctly in Snapshot', () => {
    useChangeExpirationDate.mockImplementation(
      () => DataMock_ChangeExpirationDate
    );

    const screenTest = render(
      <NavigationWrapper screen={ChangeExpirationDateScreen} />
    );
    expect(screenTest?.toJSON()).toMatchSnapshot();
  });
});
