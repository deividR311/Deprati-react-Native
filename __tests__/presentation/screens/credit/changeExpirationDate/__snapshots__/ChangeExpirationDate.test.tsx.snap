// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeExpirationDate renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RCTScrollView>
            <View>
              <View
                style={
                  Object {
                    "padding": 18,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "borderRadius": 3,
                      "elevation": 4,
                      "flexDirection": "row",
                      "padding": 14,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.25,
                      "shadowRadius": 3.84,
                    }
                  }
                >
                  <Image
                    height={64}
                    source={
                      Object {
                        "testUri": "../../../assets/images/cardCredito.png",
                      }
                    }
                    style={
                      Object {
                        "height": 64,
                        "width": 106,
                      }
                    }
                    width={106}
                  />
                  <View
                    style={
                      Object {
                        "height": 62,
                        "marginLeft": 17,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "Roboto-Regular",
                            "fontSize": 10,
                            "fontWeight": "400",
                            "letterSpacing": 1,
                          },
                          Object {
                            "color": "#000000",
                            "fontFamily": "Roboto-Medium",
                            "fontSize": 12,
                            "fontWeight": "500",
                            "letterSpacing": 0.8,
                          },
                          Object {
                            "flexGrow": 1,
                          },
                        ]
                      }
                    >
                      PATRICIA L. DE UGALDE
14XXX86 00
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "Roboto-Medium",
                            "fontSize": 12,
                            "fontWeight": "500",
                            "letterSpacing": 0.8,
                          },
                          Object {
                            "color": "#26323880",
                          },
                        ]
                      }
                    >
                      Cliente desde 2000
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#F5F6F9",
                      "marginVertical": 16,
                      "padding": 12,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Roboto-Medium",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "letterSpacing": 0.44,
                          "lineHeight": 24,
                          "marginBottom": 4,
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    Fecha de pago actual
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#263238",
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 14,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    14 de cada mes
                  </Text>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Roboto-Medium",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.44,
                        "lineHeight": 24,
                        "marginBottom": 4,
                        "textAlign": "left",
                      },
                      Object {
                        "marginBottom": 10,
                      },
                    ]
                  }
                >
                  Cambia la fecha máxima de pago
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#263238",
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 14,
                        "fontWeight": "400",
                      },
                    ]
                  }
                >
                  El día que selecciones aplicará para todos tus siguientes pagos de tu crédito De Prati
                </Text>
                <View
                  style={
                    Object {
                      "marginVertical": 20,
                    }
                  }
                >
                  <View
                    accessibilityLabel=""
                    accessible={true}
                    onLayout={[Function]}
                    testID=""
                  >
                    <View />
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#FFC70033",
                      "borderColor": "#FFC700",
                      "borderRadius": 4,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "padding": 12,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "display": "flex",
                        "justifyContent": "flex-start",
                      }
                    }
                  >
                    <RNSVGSvgView
                      align="xMidYMid"
                      bbHeight={21}
                      bbWidth={21}
                      fill="none"
                      focusable={false}
                      height={21}
                      meetOrSlice={0}
                      minX={0}
                      minY={0}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderWidth": 0,
                          },
                          Object {
                            "flex": 0,
                            "height": 21,
                            "width": 21,
                          },
                        ]
                      }
                      vbHeight={21}
                      vbWidth={21}
                      width={21}
                    >
                      <RNSVGGroup
                        fill={null}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                      >
                        <RNSVGPath
                          d="M9.02624 14.5C9.02624 13.9477 9.47526 13.5 10.0292 13.5C10.5831 13.5 11.0321 13.9477 11.0321 14.5C11.0321 15.0523 10.5831 15.5 10.0292 15.5C9.47526 15.5 9.02624 15.0523 9.02624 14.5ZM9.02624 6.5C9.02624 5.94772 9.47526 5.5 10.0292 5.5C10.5831 5.5 11.0321 5.94772 11.0321 6.5V10.5C11.0321 11.0523 10.5831 11.5 10.0292 11.5C9.47526 11.5 9.02624 11.0523 9.02624 10.5V6.5ZM10.0241 0.5C4.48303 0.5 0 4.975 0 10.5C0 16.025 4.48303 20.5 10.0241 20.5C15.5653 20.5 20.0583 16.025 20.0583 10.5C20.0583 4.975 15.5653 0.5 10.0241 0.5ZM10.0292 18.5C5.59627 18.5 2.00583 14.92 2.00583 10.5C2.00583 6.08 5.59627 2.5 10.0292 2.5C14.462 2.5 18.0525 6.08 18.0525 10.5C18.0525 14.92 14.462 18.5 10.0292 18.5Z"
                          fill={4280693304}
                          propList={
                            Array [
                              "fill",
                            ]
                          }
                        />
                      </RNSVGGroup>
                    </RNSVGSvgView>
                  </View>
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "paddingHorizontal": 12,
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "Roboto-Regular",
                            "fontSize": 14,
                            "fontWeight": "400",
                            "letterSpacing": 0.25,
                            "lineHeight": 20,
                          },
                        ]
                      }
                    >
                      <Text>
                        <Text>
                          No puedes cambiar tu fecha máxima de pago si tienes una deuda pendiente y 
                        </Text>
                      </Text>
                      <Text
                        style={
                          Object {
                            "fontFamily": "Roboto-Bold",
                            "fontWeight": "bold",
                          }
                        }
                      >
                        <Text>
                          este cambio se hará efectivo a partir del próximo mes.
                        </Text>
                        <Text
                          style={
                            Object {
                              "fontWeight": "400",
                            }
                          }
                        >
                          
                        </Text>
                      </Text>
                    </Text>
                  </View>
                </View>
              </View>
              <View
                accessibilityLabel="view_"
                accessible={true}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    Object {
                      "paddingHorizontal": 10,
                    },
                  ]
                }
                testID="view_"
              >
                <View
                  accessibilityLabel=""
                  accessibilityLiveRegion="polite"
                  accessibilityRole="checkbox"
                  accessibilityState={
                    Object {
                      "checked": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 18,
                        "height": 36,
                        "padding": 6,
                        "width": 36,
                      },
                    ]
                  }
                  testID=""
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "transform": Array [
                          Object {
                            "scale": 1,
                          },
                        ],
                      }
                    }
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 24,
                          },
                          Array [
                            Object {
                              "lineHeight": 24,
                              "transform": Array [
                                Object {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ],
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      󰄱
                    </Text>
                    <View
                      style={
                        Array [
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          },
                          Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                          },
                        ]
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "borderColor": "#000000",
                            "borderWidth": 0,
                            "height": 14,
                            "width": 14,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "textAlign": "left",
                      },
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "Roboto-Regular",
                        },
                        Object {},
                      ]
                    }
                  >
                    acceptThe
                  </Text>
                  <Text
                    onPress={[Function]}
                    style={
                      Array [
                        Object {
                          "fontFamily": "Roboto-Regular",
                          "marginHorizontal": 5,
                        },
                        Object {
                          "color": "#E31A2C",
                          "fontFamily": "Roboto-Regular",
                        },
                      ]
                    }
                  >
                    termsAndConditions
                  </Text>
                </Text>
              </View>
              <View
                style={
                  Object {
                    "marginTop": 15,
                    "paddingHorizontal": 15,
                  }
                }
              >
                <View
                  accessibilityLabel=""
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#D3D3D3",
                      "borderRadius": 5,
                      "justifyContent": "center",
                      "opacity": 1,
                      "paddingVertical": 10,
                      "width": "100%",
                    }
                  }
                  testID=""
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": undefined,
                          },
                          Object {},
                          Object {},
                        ]
                      }
                    >
                      CAMBIAR FECHA
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "height": 80,
                    "width": "100%",
                  }
                }
              />
            </View>
          </RCTScrollView>
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            onDismiss={[Function]}
            transparent={true}
            visible={false}
          />
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            onDismiss={[Function]}
            transparent={true}
            visible={false}
          />
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            onDismiss={[Function]}
            transparent={true}
            visible={false}
          />
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            onDismiss={[Function]}
            transparent={true}
            visible={false}
          />
          <Modal
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          />
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
